### PR TITLE
Fix ignoring gateway parameter

### PR DESCRIPTION
**Pull request by Kasherpete**
Simple fix. I can verify this works with no issues.

EDIT: nvm what I said about "no issues".... let me change this rly quick

### CODE REVIEW

**Comment by Kasherpete** [comment]
Alright, this last commit should work fine, I tested it a bit. Code cleanup, error clarity, and added a couple colors for the console. I think it may ignore errors from PNRgateway, I'll check that out really quick

**Comment by Kasherpete** [comment]
Should be done! May need testing, however it should be working.

File: demo.py [hunk 1]
```diff
@@ -99,10 +99,10 @@ def register(conn, users):
             print("Would you like to register a phone number? (y/n)")
             if input("> ").lower() == "y":
                 import sms_registration
-                if args.gateway is not None:
-                    sms_registration.GATEWAY = args.gateway
-                if args.phone is not None:
-                    sms_registration.PHONE_IP = args.phone
+
+                if args.phone is None:
+                    print('\033[91m', 'ERROR: You did not supply an IP.', '\033[0m')
```
Comment (JJTech0130) [inline demo.py]: Use `logging.error()` instead

File: demo.py [hunk 2]
```diff
@@ -144,9 +144,9 @@ def register(conn, users):
 
         if args.reregister:
             print("Re-registering...")
-            users = register(conn, users)
+            register(conn, users)
 
-        print(f"Done?")
+        print('\033[92m', "Done!", '\033[0m')
```
Comment (JJTech0130) [inline demo.py]: Use `logging.info()`

File: sms_registration.py [hunk 3]
```diff
@@ -8,30 +9,53 @@
...
+def register(push_token: bytes, no_parse, gateway: str | None, phone_ip: str) -> tuple[str, bytes]:
     """Forwards a registration request to the phone and returns the phone number, signature for the provided push token"""
     if gateway is None:
         print("Requesting device MCC+MNC for gateway detection...")
-        mccmnc = requests.get(f"http://{PHONE_IP}:{API_PORT}/info", timeout=30).text
+
+        try:
+            mccmnc = requests.get(f"http://{phone_ip}:{API_PORT}/info", timeout=30).text
+        except ConnectionError:
+            print('\033[91m', 'ERROR: Could not connect to device! Please open the Pypush SMS Helper App.', '\033[0m')
```
Comment (JJTech0130) [inline sms_registration.py]: logging.error

Comment (JJTech0130) [reply sms_registration.py]: also, don't exit(-1) here. Instead, throw an exception, demo.py can catch it and exit if that's the behavior we want

File: sms_registration.py [hunk 4]
```diff
@@ -8,30 +9,53 @@
...
+            exit(-1)
+
+        except Exception as e:
+            print('\033[91m', f'ERROR: An unknown error occurred: {e}', '\033[0m')
```
Comment (JJTech0130) [inline sms_registration.py]: same as above

File: sms_registration.py [hunk 5]
```diff
@@ -8,30 +9,53 @@
...
+            exit(-1)
+
         print("MCC+MNC received! " + mccmnc)
         print("Determining gateway...")
         gateway = gateway_fetch.getGatewayMCCMNC(mccmnc)
     if gateway is not None:
         print("Gateway found!  " + str(gateway))
     else:
-        print("No gateway was provided, and automatic gateway detection failed. Please run again with the --gateway flag.")
+        print('\033[91m', "ERROR: No gateway was provided, and automatic gateway detection failed. Please run again with the --gateway flag.", '\033[0m')
```
Comment (JJTech0130) [inline sms_registration.py]: logging

File: sms_registration.py [hunk 6]
```diff
@@ -8,30 +9,53 @@
...
     token = push_token.hex().upper()
     req_id = random.randint(0, 2**32)
     sms = f"REG-REQ?v=3;t={token};r={req_id};"
     print("Sending message and waiting for response...")
-    r = requests.get(f"http://{PHONE_IP}:{API_PORT}/register", params={"sms": sms, "gateway": gateway}, timeout=30)
-    print("Received response from device!")
+    try:
+        r = requests.get(f"http://{phone_ip}:{API_PORT}/register", params={"sms": sms, "gateway": gateway}, timeout=30)
+    except Exception as e:
+        print('\033[91m', f'ERROR: An unknown error occurred: {e}', '\033[0m')
```
Comment (JJTech0130) [inline sms_registration.py]: same thing about error handling and exit, and logging

File: sms_registration.py [hunk 7]
```diff
@@ -8,30 +9,53 @@
...
+        exit(-1)
+
+    if 'error:' in r.text.lower():
+        print('\033[91m', f'ERROR: An error was thrown from PNRgateway Client: {r.text}', '\033[0m')
```
Comment (JJTech0130) [inline sms_registration.py]: logging, raise an exception?

File: sms_registration.py [hunk 8]
```diff
@@ -8,30 +9,53 @@
...
+
+        # POSSIBLE ERROR FROM APP:
+
+        # Error: timeout waiting for response from gateway (unlikely, timeout is implemented here too)
+        # Error sending SMS
+        # Error: sms parameter not found
+        # Error: gateway parameter not found
+
+    print('\033[92m', "Received response from device!", '\033[0m')
```
Comment (JJTech0130) [inline sms_registration.py:66]: logging

**Comment by Kasherpete** [comment]
I can probably adjust this in the next couple days. If anyone else wants to do this rly quick it would be helpful

File: demo.py (hunk 1 above)
Comment (JJTech0130) [reply demo.py]:
```suggestion
                    logging.error("You did not supply an IP")
```

File: demo.py [hunk 9]
```diff
@@ -99,10 +99,10 @@ def register(conn, users):
...
             if input("> ").lower() == "y":
                 import sms_registration
-                if args.gateway is not None:
-                    sms_registration.GATEWAY = args.gateway
-                if args.phone is not None:
-                    sms_registration.PHONE_IP = args.phone
+
+                if args.phone is None:
+                    print('\033[91m', 'ERROR: You did not supply an IP.', '\033[0m')
+                    exit(-1)
```
Comment (JJTech0130) [inline demo.py]:
```suggestion
                    raise Exception("IP address not specified")
```

File: demo.py (hunk 2 above)
Comment (JJTech0130) [reply demo.py]:
```suggestion
        logging.info("Done!")
```

File: sms_registration.py [hunk 10]
```diff
@@ -8,30 +9,53 @@
...
     """Forwards a registration request to the phone and returns the phone number, signature for the provided push token"""
     if gateway is None:
         print("Requesting device MCC+MNC for gateway detection...")
-        mccmnc = requests.get(f"http://{PHONE_IP}:{API_PORT}/info", timeout=30).text
+
+        try:
+            mccmnc = requests.get(f"http://{phone_ip}:{API_PORT}/info", timeout=30).text
+        except ConnectionError:
+            print('\033[91m', 'ERROR: Could not connect to device! Please open the Pypush SMS Helper App.', '\033[0m')
+            exit(-1)
```
Comment (JJTech0130) [inline sms_registration.py]:
```suggestion
            raise
```

File: sms_registration.py (hunk 3 above)
Comment (JJTech0130) [reply sms_registration.py]:
```suggestion
            logger.error("Could not connect to device! Please open the pypush SMS Helper App.")
```

File: sms_registration.py [hunk 11]
```diff
@@ -8,30 +9,53 @@
 import urllib3
 urllib3.disable_warnings()
```
Comment (JJTech0130) [inline sms_registration.py:14]:
```suggestion
import logging
logger = logging.getLogger("sms_registration")
```

File: sms_registration.py (hunk 4 above)
Comment (JJTech0130) [reply sms_registration.py]:
```suggestion
            logger.error(f"An unknown error occurred: {e}")
```

**Review by JJTech0130 — COMMENTED** [review]
here are some suggestions, not all (you can click add to batch and commit all these suggestions)

**Comment by Kasherpete** [comment]
We should make some custom exceptions so we aren't logging an error and raising, or just raising a generic `Exception` with a note. I may be able to do this today as well as resolve new merge conflicts, I'll just have to sacrifice some time away from my PS4 lol

**Comment by SpaceSaver** [comment]
@Kasherpete https://github.com/Kasherpete/pypush/pull/1

**Comment by SpaceSaver** [comment]
@Kasherpete The merge conflicts are annoying, might need to manually copy over changes to the latest version of the branch

**Comment by JJTech0130** [comment]
> We should make some custom exceptions so we aren't logging an error and raising, or just raising a generic `Exception` with a note. I may be able to do this today as well as resolve new merge conflicts, I'll just have to sacrifice some time away from my PS4 lol

There’s just one conflict, shouldn’t take long. I would make the exception thing a separate PR.
EDIT: In fact, if you just leave it for now, I can fix the conflict when I merge it

**Comment by Kasherpete** [comment]
@JJTech0130 me and SpaceSaver are fixing it on my fork, when it gets fixed I'll update it here. It's really just inconsistent exceptions and logging. I'll follow everything you outlined in the review.

**Comment by Kasherpete** [comment]
> > We should make some custom exceptions so we aren't logging an error and raising, or just raising a generic `Exception` with a note. I may be able to do this today as well as resolve new merge conflicts, I'll just have to sacrifice some time away from my PS4 lol
> 
> There’s just one conflict, shouldn’t take long. I would make the exception thing a separate PR.
> EDIT: In fact, if you just leave it for now, I can fix the conflict when I merge it

You know what, I'll have some time later today. I'll open a small PR on my fork adding custom exceptions, merge SpaceSaver's PR on my fork, then bring it all here for you to review. You shouldn't have to worry about small maintenance things like this, I can take care of it unless there's something specific you'd like to implement

**Comment by JJTech0130** [comment]
Just a heads up, I'm probably going to merge the openhaystack stuff into the async branch, and then rework all of the SMS stuff entirely so that it can coexist. In the openhaystack and cloudkit branches you can see how I'm starting to structure that, I think sms_registration and interactive_demo would be 2 files in examples/. Then the other files would be refactored into a proper package and eventually published to pypi

For now though, this is good, since it will probably take me a bit to get around to all that refactoring.

**Comment by Kasherpete** [comment]
> Just a heads up, I'm probably going to merge the openhaystack stuff into the async branch, and then rework all of the SMS stuff entirely so that it can coexist. In the openhaystack and cloudkit branches you can see how I'm starting to structure that, I think sms_registration and interactive_demo would be 2 files in examples/. Then the other files would be refactored into a proper package and eventually published to pypi

Do you want me to like, PR my API branch of Pypush? So you can login and send messages with a few lines of code? Low level operations would still be very accessible, I can make sure of it within a __init__.py file

We should probably focus on merging all the branches together first though. Before too many changes are made.

**Comment by JJTech0130** [comment]
Nah, lots of things will need refactoring, I can do it myself.
And yeah, branches should definitely be merged first, just giving you a heads-up about my long term plans. (and why this PR doesn't have to be perfect, it's just a stop-gap)

**Comment by Kasherpete** [comment]
Got it. I will get the logging changes done today, and probably make an all-in-one API branch on my own fork and keep it up to date with everything you do, for a reference PR in the future

**Comment by Kasherpete** [comment]
FINALLY ready to merge

**Comment by JJTech0130** [comment]
Merging now, I'll fix the print stuff later...

**Pull request merged**